File: android/src/main/java/expo/modules/settings/ExpoSettingsModule.kt
package expo.modules.settings

import android.content.Context
import android.content.SharedPreferences
import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class ExpoSettingsModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ExpoSettings")

    Function("setContacts") { contacts: String ->
      getPreferences().edit().putString("contacts", contacts).commit()
    }

    Function("getContacts") {
      return@Function getPreferences().getString("contacts", "[]")
    }
  }

  private val context
  get() = requireNotNull(appContext.reactContext)

  private fun getPreferences(): SharedPreferences {
    return context.getSharedPreferences(context.packageName + ".settings", Context.MODE_PRIVATE)
  }
}

File: ios/ExpoSettingsModule.swift
import ExpoModulesCore

public class ExpoSettingsModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ExpoSettings")

    Function("setContacts") { (contacts: String) -> Void in
      UserDefaults.standard.set(contacts, forKey:"contacts")
    }

    Function("getContacts") { () -> String in
      UserDefaults.standard.string(forKey: "contacts") ?? "[]"
    }
  }
}
